test cases/common/250 system include dir/meson.build

```diff
@@ -0,0 +1,11 @@
+project('system_include_dir', 'cpp',
+    version : '0.1',
+    default_options : 'werror=true',
+)
+
+lib_include_directories = include_directories('lib', is_system: true)
+
+if meson.get_compiler('cpp').get_id() != 'msvc'
+    add_project_arguments('-Wsign-conversion', language: 'cpp')
+endif
```

Comment:
This test is really fragile, since it assumes every compiler that isn't msvc has -Wsign-conversion (it doesn't). It would probably be better to check this as:
```meson
if not ['gcc', 'clang',' 'clang-cl'].contains(meson.get_compiler('cpp').get_id())
  error('MESON_SKIP_TEST: compiler @0@ either doesn\'t support is_system includes or needs to have support for this test aded'.format(meson.get_compiler('cpp').get_id())
endif
```

Which means that no compilers get her unexpectedly, an dyou don't need the fragile !msvc test (which will fail on lots of compilers because for example, icc needs a different argument).

Reply:
Is this enough to statisfy CI or should I also configure the test to be skipped when compiler does not have system includes ?